code/modules/hydroponics/biogenerator.dm

```diff
@@ -167,74 +192,54 @@
 	if(processing)
 		to_chat(user, span_warning("The biogenerator is in the process of working."))
 		return
-	var/processing_time = 0
-	for(var/obj/item/food/grown/I in contents)
-		processing_time += 5
-		if(I.reagents.get_reagent_amount(/datum/reagent/consumable/nutriment) < 0.1)
-			points += 1 * productivity
-		else
-			points += I.reagents.get_reagent_amount(/datum/reagent/consumable/nutriment) * 10 * productivity
-		qdel(I)
-	if(processing_time)
+	if(biomass >= max_biomass)
+		say("Warning: The biomass storage is full!")
+		return
+	if(locate(/obj/item/food/grown) in contents)
 		processing = TRUE
+		soundloop.start()
 		update_appearance()
-		playsound(loc, 'sound/machines/blender.ogg', 50, TRUE)
-		use_power(processing_time * active_power_usage * 0.1) // .1 needed here to convert time (in deciseconds) to seconds such that watts * seconds = joules
-		sleep(processing_time + 15 / productivity)
+		var/potential_biomass = 0
+		for(var/obj/item/food/grown/I in contents)
+			potential_biomass = max(1, I.reagents.get_reagent_amount(/datum/reagent/consumable/nutriment)) * productivity
+			qdel(I)
+			if(potential_biomass)
+				while(processing && potential_biomass > 0)
+					use_power(active_power_usage * 0.1) // .1 needed here to convert time (in deciseconds) to seconds such that watts * seconds = joules
```

Comment:
```suggestion
					use_power(active_power_usage * (1 SECONDS)) // Seconds needed here to convert time (in deciseconds) to seconds such that watts * seconds = joules
```

Reply:
@Mothblocks We fucked up here, it should've been 0.01 SECONDS. Now the Biogen uses 300kw instead of 3kw. I will make a PR to fix that. I noticed this only after it got into master.